Build failure: qalculate-gtk

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build nixpkgs#qalculate-gtk`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/327168190

### Relevant log output

```console
Running phase: unpackPhase
unpacking source archive /nix/store/1rlvp4k7yfyk3j7r0djx6kc1d6x2g2w9-source
source root is source
Running phase: patchPhase
Running phase: autoreconfPhase
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: running: autopoint --force
Copying file config.rpath
Creating directory m4
Copying file m4/build-to-host.m4
Copying file m4/gettext.m4
Copying file m4/host-cpu-c-abi.m4
Copying file m4/iconv.m4
Copying file m4/intlmacosx.m4
Copying file m4/lib-ld.m4
Copying file m4/lib-link.m4
Copying file m4/lib-prefix.m4
Copying file m4/nls.m4
Copying file m4/po.m4
Copying file m4/progtest.m4
Copying file po/Makefile.in.in
Copying file po/Makevars.template
Copying file po/Rules-quot
Copying file po/boldquot.sed
Copying file po/en@boldquot.header
Copying file po/en@quot.header
Copying file po/fetch-po
Copying file po/insert-header.sed
Copying file po/quot.sed
Copying file po/remove-potcdate.sed
autoreconf: running: aclocal --force -I m4
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: configure.ac: not running autopoint a second time
autoreconf: running: aclocal --force -I m4
autoreconf: running: /nix/store/vhm22v5pqh355py36si9nfn9rifbj8is-autoconf-2.73/bin/autoconf --force
autoreconf: running: /nix/store/vhm22v5pqh355py36si9nfn9rifbj8is-autoconf-2.73/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:20: installing './compile'
configure.ac:25: installing './config.guess'
configure.ac:25: installing './config.sub'
configure.ac:11: installing './install-sh'
configure.ac:11: installing './missing'
src/Makefile.am: installing './depcomp'
autoreconf: configure.ac: not running make: --make not given
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
fixing libtool script ./ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/f700nj7wlwg441h39gkq29qbviy99sgq-bash-5.3p9/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/ccnqiynqv9halsvg4alhxs63j99szrwy-qalculate-gtk-5.10.0
checking for a BSD-compatible install... /nix/store/nixxlz2dfdwmy6r8da5sas4nrnj7sq3z-coreutils-9.10/bin/install -c
checking whether sleep supports fractional seconds... yes
checking filesystem timestamp resolution... 0.01
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/nixxlz2dfdwmy6r8da5sas4nrnj7sq3z-coreutils-9.10/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking xargs -n works... yes
checking whether UID '361' is supported by ustar format... yes
checking whether GID '350' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... no
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether clang accepts -g... yes
checking for clang option to enable C23 features... -std=gnu23
checking whether clang -std=gnu23 understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of clang -std=gnu23... none
checking whether the compiler supports GNU C++... yes
checking whether clang++ accepts -g... yes
checking dependency style of clang++... none
checking whether ln -s works... yes
checking for grep that handles long lines and -e... /nix/store/sk1la8r8sa1rlb8m9kwy5hi29f58sjd9-gnugrep-3.12/bin/grep
checking for egrep... /nix/store/sk1la8r8sa1rlb8m9kwy5hi29f58sjd9-gnugrep-3.12/bin/grep -E
checking build system type... aarch64-apple-darwin25.3.0
checking host system type... aarch64-apple-darwin25.3.0
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/k6894r7p4545j4sbsyq698pqjwa8vfkh-gnused-4.9/bin/sed
checking for fgrep... /nix/store/sk1la8r8sa1rlb8m9kwy5hi29f58sjd9-gnugrep-3.12/bin/grep -F
checking for ld used by clang -std=gnu23... ld
checking if the linker (ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking the maximum length of command line arguments... 786432
checking how to convert aarch64-apple-darwin25.3.0 file names to aarch64-apple-darwin25.3.0 format... func_convert_file_noop
checking how to convert aarch64-apple-darwin25.3.0 file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ranlib... ranlib
checking for archiver @FILE support... @
checking for strip... strip
checking command to parse nm output from clang -std=gnu23 object... ok
checking for sysroot... no
checking for a working dd... /nix/store/nixxlz2dfdwmy6r8da5sas4nrnj7sq3z-coreutils-9.10/bin/dd
checking how to truncate binary pipes... /nix/store/nixxlz2dfdwmy6r8da5sas4nrnj7sq3z-coreutils-9.10/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... no
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -no_fixup_chains linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if clang -std=gnu23 supports -fno-rtti -fno-exceptions... yes
checking for clang -std=gnu23 option to produce PIC... -fno-common -DPIC
checking if clang -std=gnu23 PIC flag -fno-common -DPIC works... yes
checking if clang -std=gnu23 static flag -static works... no
checking if clang -std=gnu23 supports -c -o file.o... yes
checking if clang -std=gnu23 supports -c -o file.o... (cached) yes
checking whether the clang -std=gnu23 linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin25.3.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking how to run the C++ preprocessor... clang++ -E
checking for ld used by clang++... ld
checking if the linker (ld) is GNU ld... no
checking whether the clang++ linker (ld) supports shared libraries... yes
checking for clang++ option to produce PIC... -fno-common -DPIC
checking if clang++ PIC flag -fno-common -DPIC works... yes
checking if clang++ static flag -static works... no
checking if clang++ supports -c -o file.o... yes
checking if clang++ supports -c -o file.o... (cached) yes
checking whether the clang++ linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin25.3.0 dyld
checking how to hardcode library paths into programs... immediate
checking for pthread_create in -lpthread... yes
checking for unordered_map... yes
checking pkg-config is at least version 0.9.0... yes
checking for
        glib-2.0 >= 2.4
        libxml-2.0
        ... yes
checking for
        gtk+-3.0 >= 3.10
        gdk-pixbuf-2.0
        ... yes
checking for gtk-mac-integration-gtk3 >= 3.0.1... no
configure: error: Package requirements (gtk-mac-integration-gtk3 >= 3.0.1) were not met:

No package 'gtk-mac-integration-gtk3' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables GTK_MAC_CFLAGS
and GTK_MAC_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```

### Additional context

_No response_

### System metadata

- system: `"aarch64-darwin"`
 - host os: `Darwin 25.5.0, macOS 26.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Determinate Nix 3.8.2) 2.30.1`
 - nixpkgs: `/nix/store/jzd9vz08h5b7s3rgxzpw28mpw4rxk2ra-source`

### Notify maintainers

@doronbehar @cyclic-pentane @Aleksanaa

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/qa/qalculate-gtk/package.nix
+++ b/pkgs/by-name/qa/qalculate-gtk/package.nix
@@ -2,10 +2,12 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   autoreconfHook,
   pkg-config,
   libqalculate,
   gtk3,
+  gtk-mac-integration-gtk3,
   curl,
   wrapGAppsHook3,
   desktopToDarwinBundle,
@@ -22,6 +24,16 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-JZfolSRLRLtv529f25lEPYOlz+y+EdRqKA0Y5d1dK3s=";
   };
 
+  patches = [
+    # Is meaningfull only for Darwin but is better applying unconditionally.
+    # Can be removed in 5.11 probably, see:
+    # https://github.com/Qalculate/qalculate-gtk/pull/705
+    (fetchpatch {
+      url = "https://github.com/Qalculate/qalculate-gtk/commit/2f1b6a32d98f122d44d158c20f79a5b9aaf9669f.patch";
+      hash = "sha256-v9T8wWvqlI9l6BUszf/575qmoFvh88Cdj/m2XqV8Q4k=";
+    })
+  ];
+
   hardeningDisable = [ "format" ];
 
   nativeBuildInputs = [
@@ -35,6 +47,9 @@ stdenv.mkDerivation (finalAttrs: {
     libqalculate
     gtk3
     curl
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    gtk-mac-integration-gtk3
   ];
 
   enableParallelBuilding = true;

--- a/pkgs/by-name/qa/qalculate-gtk/package.nix
+++ b/pkgs/by-name/qa/qalculate-gtk/package.nix
@@ -30,11 +30,13 @@ stdenv.mkDerivation (finalAttrs: {
     wrapGAppsHook3
   ]
   ++ lib.optionals stdenv.hostPlatform.isDarwin [ desktopToDarwinBundle ];
+
   buildInputs = [
     libqalculate
     gtk3
     curl
   ];
+
   enableParallelBuilding = true;
 
   meta = {

